### PR TITLE
Set application key in .env file (if exists) or fallback config file.

### LINKED ISSUE #29: Add support for setting the application key in .env files when running key:generate
See:
- octobercms/october#5496
- octobercms/library#560
- octobercms/october#5514

### PR DESCRIPTION
**Pull request by ericp-mrel**
Original PR: https://github.com/octobercms/library/pull/560

Fixes: https://github.com/wintercms/winter/issues/29

### CODE REVIEW

File: tests/fixtures/config/sample-config-with-env.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    | Application Timezone
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify the default timezone for your application, which
+    | will be used by the PHP date and date-time functions. We have gone
+    | ahead and set this to a sensible default for you out of the box.
+    |
+    */
+
+    'timezone' => "October's time",
```
Comment (LukeTowers) [inline tests/fixtures/config/sample-config-with-env.php]:
```suggestion
    'timezone' => "UTC",
```

**Comment by bennothommo** [comment]
@ericp-mrel there is a PR for a complete `ConfigWriter` rewrite (see https://github.com/wintercms/storm/pull/40), which supersedes this. I'll get @jaxwilko to include your fix in that PR and merge it, and ensure you get some credit for the fix.

Thanks for your work on this.

**Comment by ericp-mrel** [comment]
@bennothommo Will the new ConfigWriter re-write know to update the .env file variables? I think a PR/changes to the key generate command are still needed in order for the key to be set in the .env file (if it exists) after the Config Writer re-write, because I'm assuming that class will still only be responsible for managing PHP config files.

**Comment by bennothommo** [comment]
@ericp-mrel it might be ideal for ConfigWriter to handle the .env file as well, as .env is ostensibly a configuration file, but @jaxwilko might be able to clarify that further.

**Comment by jaxwilko** [comment]
@ericp-mrel @bennothommo my plan is to add an EnvFile class with a similar interface to ConfigFile, then have support for (but not necessarily defaulting to) updating env values when updating a config file that has a env() call.

The end goal is to replace the ConfigWriter usages, especially the winter:install command which I've been using for testing.

There is a PR open for ConfigFile so feel free to take a look and add any comments you have 👍.

If there's any more info I can give feel free to ping me on here or on discord :)

**Pull request closed without merging**